packages/pwa-kit-react-sdk/src/ssr/universal/utils.js

```diff
@@ -51,3 +51,64 @@ export const getProxyConfigs = () => {
     // Clone to avoid accidental mutation of important configuration variables.
     return configs.map((config) => ({...config}))
 }
+
+/**
+ * Returns the express app configuration file in object form. The object resolution will
+ * be as follows (from highest to lowest priority):
+ *
+ * > {target_name}.ext
+ * > local.ext
+ * > default.ext
+ * > package.json (mobify.key)
+ *
+ * Each file marked with `ext` can optionally be terminated with `yml`, `yaml` or
+ * `json` in that priority.
+ *
+ * NOTE: This is an isomorphic function, when run on the browser the config returned will
+ * be the value serialized in the html.
+ *
+ * @returns - the application configuration object.
+ */
+/* istanbul ignore next */
+export const getConfig = () => {
+    if (onClient) {
+        return (
+            window.__CONFIG__ ||
+            JSON.parse(document.getElementById('mobify-data').innerHTML).__CONFIG__
+        )
+    }
+
+    const isRemote = Object.prototype.hasOwnProperty.call(process.env, 'AWS_LAMBDA_FUNCTION_NAME')
+    let moduleName = process?.env?.DEPLOY_TARGET || ''
+
+    // Synchronously load the `cosmiconfig` so we don't get errors from webpack in the
+    // browser bundle.
+    const _require = eval('require')
+    const {cosmiconfigSync} = _require('cosmiconfig')
+
+    // Match config files based on the specificity from most to most general.
+    const explorerSync = cosmiconfigSync(moduleName, {
+        packageProp: 'mobify',
+        searchPlaces: [
+            `config/${moduleName}.yml`,
+            `config/${moduleName}.yaml`,
+            `config/${moduleName}.json`,
+            `config/local.yml`,
+            `config/local.yaml`,
+            `config/local.json`,
+            `config/default.yml`,
+            `config/default.yaml`,
+            `config/default.json`,
+            `package.json`
+        ].map((path) => (isRemote ? `build/${path}` : path))
+    })
+
+    // Load the config synchronously using a custom "searchPlaces".
+    const {config} = explorerSync.search() || {}
+
+    if (!config) {
+        throw new Error('Application configuration not found!')
```

Comment:
nit, could we improve the error message to include the specific file locations like:

```js
throw new Error(`Application configuration not found!

Possible configuration file locations:
config/${moduleName}.yml
config/${moduleName}.yaml
config/${moduleName}.json
config/local.yml
config/local.yaml
config/local.json
config/default.yml
config/default.yaml
config/default.json
package.json
`)
```

Preview of the error:
![Screen Shot 2022-02-22 at 1 24 16 PM](https://user-images.githubusercontent.com/10948652/155221927-adbe38dd-778c-4c89-a7ca-a4aff11af2f3.png)

Reply:
Good idea. I've added this.